src/logger.ts

```diff
@@ -0,0 +1,55 @@
+import minimist from "minimist";
+import winston from "winston";
+
+export interface CustomLogger extends winston.Logger {
+    maker: winston.LeveledLogMethod;
+    taker: winston.LeveledLogMethod;
+    data: winston.LeveledLogMethod;
+}
+
+const colorizer = winston.format.colorize({
+    all: true,
+    colors: {
+        error: "red",
+        maker: "cyan",
+        taker: "yellow",
+        data: "grey",
+    },
+});
+
+export function createLogger() {
+    const level = minimist(process.argv.slice(2), {
+        default: {
+            loglevel: "info",
+        },
+    }).loglevel;
+
+    if (level !== "info" && level !== "data") {
+        throw new Error(
+            `[error] Invalid log level: ${level}. Choose one from "info" or "data"`
+        );
+    }
+
+    return winston.createLogger({
+        levels: {
+            error: 0,
+            maker: 1,
```

Comment:
Instead of abusing log levels here, I'd suggest to use customize the `format` based on a parameter that was passed in. You can then create an instance specific logger for maker and taker and don't have to pass it on every invocation.

It would also have the advantage of maker and taker being able to log on different levels :)
Can do this together in a follow-up PR :)

Reply:
It's probably abusing log levels, yep. Probably best to deal with it in a follow-up PR as you said :relaxed: